cobalt/persistent_storage/persistent_settings.cc

```diff
@@ -113,47 +113,49 @@ bool PersistentSettings::GetPersistentSettingAsBool(const std::string& key,
                                                     bool default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_bool()) return result->GetBool();
+  absl::optional<bool> result = persistent_settings.FindBool(key);
+  if (result) return *result;
   return default_setting;
 }
 
 int PersistentSettings::GetPersistentSettingAsInt(const std::string& key,
                                                   int default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_int()) return result->GetInt();
+  absl::optional<int> result = persistent_settings.FindInt(key);
+  if (result) return *result;
   return default_setting;
 }
 
 double PersistentSettings::GetPersistentSettingAsDouble(
     const std::string& key, double default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_double()) return result->GetDouble();
+  absl::optional<double> result = persistent_settings.FindDouble(key);
+  if (result) return *result;
   return default_setting;
 }
 
 std::string PersistentSettings::GetPersistentSettingAsString(
     const std::string& key, const std::string& default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_string()) return result->GetString();
+  const std::string* result = persistent_settings.FindString(key);
+  if (result) return *result;
   return default_setting;
 }
 
 std::vector<base::Value> PersistentSettings::GetPersistentSettingAsList(
     const std::string& key) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  base::Value* result = persistent_settings.Find(key);
-  if (result && result->is_list()) {
-#ifndef USE_HACKY_COBALT_CHANGES
-    return std::move(result).TakeList();
-#endif
+  const base::Value::List* result = persistent_settings.FindList(key);
+  if (result) {
+    std::vector<base::Value> values;
+    for (const auto& value : std::move(*result)) {
+      values.emplace_back(value.Clone());
```

Comment:
Here, do we end up taking ownership of the list instead of creating a duplicate that lasts? I think the duplicate gets destroyed at the end of this scope which seems fine, but want to check

Reply:
I'm not very sure what is the expected behavior of this function. It's only called by persistent_settings_test.cc, and no real use case by Cobalt. My best guess is we don't want to take the ownership but only create a duplicate? I won't worry about this that much since it's not used anyway.